### PR TITLE
Fix for addons id param to conform with other controllers

### LINKED ISSUE #6719: [API doc] Fix addonApiController idParam not being conform with the other controllers
Addon idParameter should probably be named "id" like everything else.
screen shot 2018-02-27 at 2 41 31 pm

### PR DESCRIPTION
**Pull request by cchabilall83**
fixes #6719

Addon idParameter was addonID instead of id

We changed the rename $addonID to $id to resolve the display issue

### CODE REVIEW

File: applications/dashboard/controllers/api/AddonsApiController.php [hunk 1]
```diff
@@ -178,17 +178,17 @@ public function index(array $query) {
...
      * @throws \Garden\Web\Exception\NotFoundException Throws an exception when the addon is not found.
      */
-    public function get($addonID, array $query) {
+    public function get($id, array $query) {
         $this->permission('Garden.Settings.Manage');
 
-        $id = $this->schema([
-            'addonID:s' => 'The ID of the addon.'
-        ], 'in')->setDescription('Get an addon.');
+//        $id = $this->schema([
```
Comment (DaazKu) [inline applications/dashboard/controllers/api/AddonsApiController.php]: Wut? Why did you comment the schema?

Comment (cchabilall83) [reply applications/dashboard/controllers/api/AddonsApiController.php]: I forgot to uncomment it. I thought it wasn't used anywhere, and it was  overwriting the $id param ($addonID) right a way.

Comment (DaazKu) [reply applications/dashboard/controllers/api/AddonsApiController.php]: Yeah. You need to rename addonID to id.

**Comment by tburry** [comment]
To try and explain a bit here. I put the constraint on parameters named `$id` for a reason and that is to [remove ambiguities when paths get nested](https://docs.vanillaforums.com/developer/apiv2/resource-routing/#parameter-ambiguities). I put in specific unit tests for those cases and these changes broke them.

Also keep in mind that the constraints worked the way they did because the code was done before we were at PHP7. I had relaxed the constraint in #6922 now that we have type hints. However, there are still ambiguities when `$id` parameters are strings now, but that is still backwards compatible because previously `$id` parameters could only be natural numbers. Endpoints that require an `$id` string parameter will have to be careful with what they support on sub paths.

**Comment by linc** [comment]
Also, this was not put in this sprint and I don't recall it being discussed at the sprint planning meeting.

**Comment by DaazKu** [comment]
@linc It was from last sprint and we were blocked because $id could not be a string.

**Pull request closed without merging**